src/sentry/api/endpoints/organization_group_index.py

```diff
@@ -107,11 +107,18 @@ def get(self, request, organization):
             # check to see if we've got an event ID
             if len(query) == 32:
                 groups = list(
-                    Group.objects.filter_by_event_id(
-                        project_ids,
-                        query,
-                    )
+                    Group.objects.filter_by_event_id(project_ids, query)
                 )
+                if len(groups) == 1:
+                    response = Response(
+                        serialize(
+                            groups, request.user, serializer(
+                                matching_event_id=query.strip()
```

Comment:
Just being pedantic, but we probably don't need the strip here if we know the query is len 32 and we have a direct hit